pkg/authorization/authorizer/personal_subjectaccessreview.go

```diff
@@ -11,11 +11,24 @@ import (
 )
 
 func IsPersonalAccessReview(a AuthorizationAttributes) (bool, error) {
-	req, ok := a.GetRequestAttributes().(*http.Request)
-	if !ok {
-		return false, errors.New("expected request, but did not get one")
+	switch extendedAttributes := a.GetRequestAttributes().(type) {
+	case *http.Request:
+		return isPersonalAccessReviewFromRequest(a, extendedAttributes)
+
+	case *authorizationapi.SubjectAccessReview:
+		return isPersonalAccessReviewFromSAR(extendedAttributes), nil
+
+	case *authorizationapi.LocalSubjectAccessReview:
+		return isPersonalAccessReviewFromLocalSAR(extendedAttributes), nil
+
+	default:
+		return false, fmt.Errorf("unexpected type: %v", extendedAttributes)
```

Comment:
Error message needs to be better so we can find the source of an error - "unexpected request attributes for checking personal access review"